### ISSUE TITLE
Using contract upgradability as a severity mitigation

### ISSUE THREAD

**Issue by alcueca** [code-423n4/rulebook#11]
This contract has a vulnerability in which assets can get locked in the smart contract. This would be a Severity 3.

However, the contract is upgradeable, and the sponsor argues that if the vulnerability wouldn't have been found, they could still upgrade the contract and rescue the funds. On that basis, they suggest the Severity should only be 2.

I'm inclined to mark it as Severity 3. Even if there is a way to make users whole, it will come at a high reputational cost for the protocol, and there is the risk that the rescue doesn't work through a botched upgrade or loss of control.

**Comment by GalloDaSballo** [comment]
I agree that upgradeability can't be used as an excuse to reduce the severity of a finding.

The finding is based on the code under scope, a change of the code would be outside.

The sponsor has a right to claim they will fix it after, just like they can claim they disagree with the finding or that they wontfix.

**Comment by KnownFactC4** [comment]
Per the Autumn 2023 C4 Supreme Court verdicts, the Supreme Court's verdict on this issue is: 

>Contract upgradability should never be used as a severity mitigation, i.e. we assume contracts are non-upgradable.

Link to verdict: https://docs.google.com/document/d/1Y2wJVt0d2URv8Pptmo7JqNd0DuPk_qF9EPJAj3iSQiE/edit#heading=h.anqqiguzy36l
